src/Stratis.FederatedPeg.Features.FederationGateway/Wallet/FederationWalletManager.cs

```diff
@@ -1093,9 +1063,16 @@ public void ImportMemberKey(string password, string mnemonic, string passphrase)
             this.Wallet.EncryptedSeed = encryptedSeed;
             this.SaveWallet();
 
+            this.isFederationActive = extendedKey.PrivateKey.PubKey.ToHex() == this.federationGatewaySettings.PublicKey;
+
             this.logger.LogTrace("(-)");
         }
 
+        public bool IsFederationActive()
```

Comment:
Maybe it can become a property now, but that is a detail

Reply:
Right.